File: tobby/3.heap/02.java
import java.util.PriorityQueue;
import java.util.Collections;

class Solution {
    public int solution(int stock, int[] dates, int[] supplies, int k) {
        int answer = 0;
        PriorityQueue<Integer> pq= new PriorityQueue<Integer>(Collections.reverseOrder());
        int n = 0;
        int l = dates.length-1;

        for(int i =0 ; i < k ; i++){
            if(dates[n] == i){
                pq.add(supplies[n]);
                if (n <l)
                    n++;
            }
            if (stock==0){
                stock += pq.remove();
                answer++;
            }
            stock--;
        }
        return answer;
    }
}

File: tobby/2.stackQ/02.py
def solution(p, l):
    answer = 0

    while True:
        t = p.pop(0)
        if len(p) != 0 and t < max(p):
            p.append(t)
            if l==0:
                l=len(p)
        else:
            answer +=1
            if l==0:
                return answer

        l -=1
    return answer
